crates/ruff_python_parser/src/parser/mod.rs

```diff
@@ -457,46 +458,53 @@ impl<'src> Parser<'src> {
         &self.source[range - self.tokens_range.start()]
     }
 
-    fn parse_list<T>(
+    /// Parses a list of elements into a vector where each element is parsed using
+    /// the given `parse_element` function.
+    fn parse_list_into_vec<T>(
         &mut self,
-        kind: RecoveryContextKind,
+        recovery_context_kind: RecoveryContextKind,
         parse_element: impl Fn(&mut Parser<'src>) -> T,
     ) -> Vec<T> {
         let mut elements = Vec::new();
-
-        self.parse_sequence(kind, |p| elements.push(parse_element(p)));
-
+        self.parse_list(recovery_context_kind, |p| elements.push(parse_element(p)));
         elements
     }
 
-    fn parse_sequence(
+    /// Parses a list of elements where each element is parsed using the given
+    /// `parse_element` function.
+    ///
+    /// The difference between this function and `parse_list_into_vec` is that
+    /// this function does not return the parsed elements. Instead, it is the
+    /// caller's responsibility to handle the parsed elements. This is the reason
+    /// that the `parse_element` parameter is bound to [`FnMut`] instead of [`Fn`].
+    fn parse_list(
```

Comment:
Nit: I'm happy to go with what we have today. I just want to throw this out here in case you like this more: How about `parse_list` (parses into a vec) and `parse_list_items` (only parses the items). Although the difference is probably to subtle.